File: core/src/main/java/com/pluxity/cctv/entity/Cctv.java
package com.pluxity.cctv.entity;

import com.pluxity.cctv.dto.CctvUpdateRequest;
import com.pluxity.device.entity.Device;
import com.pluxity.device.entity.DeviceCategory;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@DiscriminatorValue("cctv")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Cctv extends Device {

    private String name;

    @Column(length = 1000)
    private String url;

    @Builder
    public Cctv(String id, String name, String url, DeviceCategory category) {
        super(id, category);
        this.name = name;
        this.url = url;
    }

    public void updateCctv(CctvUpdateRequest request) {
        this.name = request.name();
        this.url = request.url();
    }

    @Override
    public String getName() {
        return this.name;
    }
}

File: core/src/main/java/com/pluxity/cctv/entity/Cctv.kt
package com.pluxity.cctv.entity

import com.pluxity.cctv.dto.CctvUpdateRequest
import com.pluxity.device.entity.Device
import com.pluxity.device.entity.DeviceCategory
import jakarta.persistence.Column
import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity

@Entity
@DiscriminatorValue("cctv")
class Cctv(
    id: String,
    category: DeviceCategory?,
    // DB 컬럼은 name 유지
    @Column(name = "name")
    var cctvName: String = "",
    @Column(length = 1000)
    var url: String?,
) : Device(id, category) {
    fun updateCctv(request: CctvUpdateRequest) {
        this.cctvName = request.name
        this.url = request.url
    }

    override fun getName(): String {
        return this.cctvName
    }
}
